text_adventure.py

```diff
@@ -0,0 +1,190 @@
+import random
+
+class GameObject:
+    verb_dict = {
+        "examine": "Examine a creature in the room.",
+        "hit": "Hit a creature in the room.",
+        "help": "Get help about the available commands.",
+        "heal": "Heal for an amount of damage.",
+        "kill": "Immediately kill the target.",
+        "status": "Get your own status."
+    }
+
+    @staticmethod
+    def get_input():
+        command = input(": ").split()
+        verb_word = command[0]
+        if verb_word in GameObject.verb_dict:
+            verb = getattr(GameObject, verb_word)
+        else:
+            print(f"Unknown verb {verb_word}")
+            print("For a list of commands type \"help\"")
+            return
+    
+        if len(command) >= 2:
+            noun_word = command[1]
+            print(verb(noun_word))
+        else:
+            print(verb("nothing"))
+
+    @staticmethod
+    def hit(noun):
+        for creature in character.current_room.creatures:
+            if creature.name == noun:
+                character.attack(creature)
+                if creature.health <= 0:
+                    character.current_room.remove_creature(creature)
+                    return f"You have killed the {creature.name}!"
+                else:
+                    creature.attack(character)
+                    return f"You prepare for your next move."
+        else:
+            return f"There is no {noun} to hit."
+    
+    @staticmethod
+    def examine(noun):
+        if noun == character.current_room.name or noun == "room":
+            return character.current_room.contents
+        for creature in character.current_room.creatures:
+            if creature.name == noun:
+                return creature.get_desc()
+        return f"There is no {noun} here."
+    
+    @staticmethod
+    def help(noun):
+        if noun == "nothing":
+            return list(GameObject.verb_dict.keys())
+        elif noun in GameObject.verb_dict:
+            return GameObject.verb_dict[noun]
+        else:
+            return f"Unknown command {noun}"
+    
+    @staticmethod
+    def heal(noun):
+        character.heal()
+        if len(character.current_room.creatures) > 0:
+            for creature in character.current_room.creatures:
```

Comment:
so this is your turn loop right? its an event you want happening, it may be more easy to reason about if you factored out a turn loop and added events in sequence to that turns event queue, instead of sequencing within event dispatch.